src/api/products/products.service.js

```diff
@@ -10,7 +13,37 @@ async function getById({ id }) {
   return product;
 }
 
+async function getRecommended({ userId }) {
+  const userOrders = await ordersService.getOrdersByUserId({ userId });
+  if (!(userOrders && userOrders.length >= 1)) {
+    // Retornamos los producto más recientes si no hay pedidos todavía
+    const products = await productsRepository.getAll();
+    return products;
+  }
+  // Obtener la ultima orden
+  const lastOrder = userOrders.slice(0, 1)[0];
```

Comment:
En caso de que elijáis devolver un error en vez de todos los productos, sería más óptimo crear un método ordersService.getLastByUserId en vez de traer todas las compras del cliente y hacer slice luego.

Además, al usar getAll en la linea 20 no **necesariamente** tiene que ser la última la que esté en la posición 0. Sería mejor usar getOrderedByDate() o algo así.

Reply:
en principio getOrdersByUserId ordena la compras de mas a menos reciente, por lo que la primera siempre es la mas reciente